### DISCUSSION TITLE
ESXi version update on a given host - govmomi call to support updates

### DISCUSSION THREAD

**Discussion by SaleemHP** [vmware/govmomi#3147] [Community Help] [answered]
Hi -
I was wondering if we can use Vim25 calls to perform ESXi update on a given host ?
Looking into these github repos I could not find anything suitable and direct usage for ESXi upgrade 

This is the command Im planning to use. i would like to know if there is an equivalent govmomi call available for ESXi update operation. 
Update Command: "/usr/bin/esxcli software profile update -d /vmfs/volumes/guid/vmware-depot.zip -p <esxi-profile>"

**Comment by dougm** [comment]
You can invoke esxcli from govmomi and govc: https://github.com/vmware/govmomi/blob/main/govc/USAGE.md#hostesxcli

**Comment by SaleemHP** [comment]
Thanks Doug. 
Isn't there a GOVMOMI API call available instead of GOVC CLI call ? Just like how we have for Maintenance Mode operations

**Reply by dougm** [reply, marked as answer]
Yes, you can use the same API that govc/host/esxcli.go uses, for example:

```go
		m := view.NewManager(c)
		kind := []string{"HostSystem"}

		v, err := m.CreateContainerView(ctx, c.ServiceContent.RootFolder, kind, true)
		if err != nil {
			return err
		}

		hosts, err := v.Find(ctx, kind, property.Filter{})
		if err != nil {
			return err
		}

		for _, ref := range hosts {
			host := object.NewHostSystem(c, ref)

			e, err := esxcli.NewExecutor(c, host)
			if err != nil {
				return err
			}

			res, err := e.Run([]string{"software", "profile", "update", "-d", "/vmfs/volumes/guid/vmware-depot.zip", "-p"})
			if err != nil {
				return err
			}

			for k, v := range res.Values[0] {
				fmt.Printf("%s=%s\n", k, v)
			}
		}
```

**Comment by SaleemHP** [comment]
Dear Doug ..
Thanks let me check and get back

**Comment by SaleemHP** [comment]
Thanks Doug
Was able to send the command to the ESXi host for setting Maintenance Mode operation. Since this is a async call I was expecting a TASK_ID to be returned back to me in the response. But I could not see any object giving me back TASK_ID, in the below struct.
type Response struct {
    Info   *CommandInfoMethod
    Values []Values
    String string
}  

When we looks at examples we get to see :
  govc tasks
  govc tasks -f
  govc tasks -f /dc1/host/cluster1`

Do we need to implement task_infra as well in our code ?
